src/Sarif/FileRegionsCache.cs

```diff
@@ -274,23 +275,48 @@ private void PopulateCharLength(NewLineIndex lineIndex, Region region)
             {
                 region.CharLength = charLength;
             }
-            SarifUtilities.DebugAssert(region.CharLength == charLength);
+            Assert(region.CharLength == charLength);
         }
 
         private NewLineIndex GetNewLineIndex(Uri uri, out string fileText)
         {
-            fileText = null;
+            Tuple<string, NewLineIndex> entry = _cache[uri.LocalPath];
+         
+            fileText = entry.Item1;
+            return entry.Item2;
+        }
+
+        private Tuple<string, NewLineIndex> BuildIndexForFile(string localPath)
+        {
```

Comment:
It might be worth a function-level comment that this is the builder function for the cache.

This is so elegant, I love it.

Reply:
Good idea. Added a comment here and where it's referenced in the ctor.